src/react_native/linear_gradient.cljs

```diff
@@ -1,6 +1,39 @@
 (ns react-native.linear-gradient
   (:require
     ["react-native-linear-gradient" :default LinearGradient]
-    [reagent.core :as reagent]))
+    [clojure.string :as string]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]
+    [taoensso.timbre :as log]))
 
-(def linear-gradient (reagent/adapt-react-class LinearGradient))
+(def ^:private linear-gradient* (reagent/adapt-react-class LinearGradient))
+
+(defn- valid-color? [color]
+  (or (keyword? color)
+      (and (string? color)
+           (or (string/starts-with? color "#")
+               (string/starts-with? color "rgb")))))
+
+(defn- split-valid-colors [acc idx color]
+  (let [color? (valid-color? color)]
+    (cond-> acc
+      :always      (update :safe-colors conj (if color? color "transparent"))
+      (not color?) (update :wrong-colors conj [idx color]))))
+
+(defn linear-gradient [props & children]
+  (assert (vector? (:colors props)))
+  (let [{:keys [wrong-colors safe-colors]} (rn/use-memo
+                                            (fn []
+                                              (reduce-kv split-valid-colors
+                                                         {:safe-colors  []
+                                                          :wrong-colors {}}
+                                                         (:colors props)))
+                                            [(:colors props)])]
+    (when (seq wrong-colors)
+      (log/error "Invalid color values in vector passed to Linear Gradient:"
+                 (reduce-kv (fn [s idx color]
+                              (str s "Index: " idx ", color: " (prn-str color)))
+                            "\n"
+                            wrong-colors)))
+    (into [linear-gradient* (assoc props :colors safe-colors)]
+          children)))
```

Comment:
The new `linear-gradient` component. It is used exactly as before.

- It checks that the colors are given in a vector (due to the validations done on the vector).
- Checks that the colors are valid in a memoized callback, since this component might be animated, we don't want to run these validations if the colors don't change.
- Logs an error when there are invalid colors.

Reply:
It looks as if most of the complication in the code is to check if the color is valid and report on invalid colors. This I think would be better handled by a schema because in release builds it's unnecessary to do these defensive checks and instrumentation is disabled in release builds. Another option if you don't want to go with the schema is to use `(when ^:boolean js/goog.DEBUG)` so the checks are elided in release builds.

What I'm suggesting is that we assume colors are correct in production and separate the code that does the runtime checks so that it's flagged behind `^:boolean js/goog.DEBUG`.

Reply:
@ilmotta 
I'm not sure if we really want to do it.

The advantage of checking the colors on runtime is to be able to skip the exception that opened the current issue, it's better for a user that the app looks a bit different in terms of the color than completely crashes.

These exceptions have happened unexpectedly on release builds and this wrapper omits the exceptions without make them completely silent.

Reply:
Thank you @ulisesmac, it's clearer now the intention of the code.

>  it's better for a user that the app looks a bit different in terms of the color than completely crashes.

100%, the app should never crash if possible.

> These exceptions have happened unexpectedly on release builds and this wrapper omits the exceptions without make them completely silent.

Just to confirm, the code you pushed now won't throw on unexpected colors? Because I completely agree with you that shouldn't happen.

Reply:
@ilmotta 

I've pushed changes again, the implementation doesn't crash if wrong colors are passed it only logs an error. This is the behavior for both debug and prod since we don't want to find unexpected crashes on release builds.